### PR TITLE
[SSDK-499] Update to Xcode 14.1

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description
- Fixes [SSDK-499](https://mapbox.atlassian.net/browse/SSDK-499)
- [Xcode 14.1 is the minimum required version for App Store submissions](https://developer.apple.com/news/?id=jd9wcyov)
- Update .swift-version to 5.7.1
- Update documentation
- See also https://xcodereleases.com/

### Checklist
- [x] Update `CHANGELOG`

[SSDK-499]: https://mapbox.atlassian.net/browse/SSDK-499?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by aokj4ck** [comment]
@mapbox/search-sdk this PR is required to fix the build in https://github.com/mapbox/mapbox-search-ios/pull/140

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ Guide: https://keepachangelog.com/en/1.0.0/
 
 ## Unreleased
 
+- [Core] Updated to Xcode 14.1 minimum version
+- [Core] Updated deployment target to iOS 14
```
Comment (kried) [inline CHANGELOG.md:10]: Is it really necessary to increase the deployment target to iOS 14? E.g. [Maps SDK uses 12.0](https://github.com/mapbox/mapbox-maps-ios/blob/main/Package.swift#L15), [Navigation v2 SDK uses also 12.0](https://github.com/mapbox/mapbox-navigation-ios/blob/2d425ca123ddeaeff33b5a26cd5c438dacce7b5c/Package.swift#L9).

Btw, the minimum iOS in [Package.swift](https://github.com/mapbox/mapbox-search-ios/blob/bd4b2642d3c4e3de4a864eacc23ed4ad6e59482c/Package.swift#L15) is still iOS 11

**Review by globaltrouble — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (aokj4ck) [reply CHANGELOG.md]: Thank you! Okay yes that makes more sense to stay consistent with the other SDKs and requiring Xcode 14.

**Review by 4rtzel — APPROVED** [review]

**Review by Udumft — APPROVED** [review]

File: Package.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ let commonMaxVersion = Version("24.0.0")
 let package = Package(
     name: "MapboxSearch",
     defaultLocalization: "en",
-    platforms: [.iOS(.v11), .macOS(.v10_15)],
+    platforms: [.iOS(.v12), .macOS(.v10_15)],
```
Comment (kried) [inline Package.swift:15]: It looks like it should be mentioned in the CHANGELOG.md

**Review by kried — APPROVED** [review]

**Pull request merged**